### PR TITLE
Daily synchronization with upstream

### PR DESCRIPTION
**Pull request by zregvart**
This adds synchronization of the `main` branch against
`tektoncd/chains:main` and `poc-tep-84` against `main`.

### CODE REVIEW

**Comment by zregvart** [comment]
Our [poc-tep-84](https://github.com/hacbs-contract/chains/tree/poc-tep-84) branch is [conflicting](https://github.com/tektoncd/chains/compare/main...hacbs-contract:poc-tep-84) with [upstream/main](https://github.com/tektoncd/chains/tree/main/) branch, [this](https://github.com/zregvart/chains/runs/6158309250) is how that would look like if this was merged.

The conflict is trivial in `vendor/modules.txt`.

File: .github/workflows/daily.yaml [hunk 1]
```diff
@@ -61,3 +61,31 @@ jobs:
...
+        with:
+          fetch-depth: 0
+
+      - name: Synchronize main
+        run: |
+          git config --global user.name 'HACBS EC Robot'
+          git config --global user.email 'hacbsec@users.noreply.github.com'
+          git remote add upstream https://github.com/tektoncd/chains.git
+          git fetch upstream main
+          git switch main
```
Comment (simonbaird) [inline .github/workflows/daily.yaml:83]: TIL about `git switch`

File: .github/workflows/daily.yaml [hunk 2]
```diff
@@ -61,3 +61,31 @@ jobs:
...
+          git rebase upstream/main
+          git push --force origin HEAD:main
+
+      - name: Synchronize poc-tep-84
+        run: |
+          git switch poc-tep-84
+          git rebase main
+          git push --force origin HEAD:poc-tep-84
```
Comment (simonbaird) [inline .github/workflows/daily.yaml:91]: @lcarva @bcaton85 what do you think?

Comment (bcaton85) [reply .github/workflows/daily.yaml:91]: I think this is good, it solves the issue with our fork falling behind, as @lcarva is experiencing. @zregvart What happens if this rebase fails due to conflicts?

Comment (zregvart) [reply .github/workflows/daily.yaml:91]: A failure [like this](https://github.com/zregvart/chains/runs/6158309250) one would happen.

**Comment by zregvart** [comment]
The value would be in building and publishing the `tep-84-poc` branch so we'd be up to date and see any issues earlier. Though we're not using builds from this fork at all at the moment. This is me thinking about continuous integration, in the sense that we'll get early warnings when the fork and upstream diverge.

**Comment by lcarva** [comment]
Let's do this. Our POC is probably going to be more long lived than I initially anticipated.

**Review by lcarva — APPROVED** [review]

**Comment by zregvart** [comment]
Too bad about the POC though.

**Pull request merged**